### ISSUE TITLE
Stokhos:  host device warning for MP Vector initializer list constructor

### ISSUE THREAD

**Issue by maartenarnst** [trilinos/Trilinos#11337]
## Bug Report
@trilinos/stokhos

### Description

When we compile Trilinos with ifpack2 and stokhos enabled on cuda, the compiler emits __host__ __device__ warnings:

```
Building CXX object packages/stokhos/src/CMakeFiles/stokhos_ifpack2_mp_16_serial.dir/Ifpack2_ContainerFactory_MP_Vector_16_Serial.cpp.o
packages/kokkos-kernels/src/sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp(2242): warning #20014-D: calling a __host__ function from a __host__ __device__ function is not allowed
packages/kokkos-kernels/src/sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp(2362): warning #20014-D: calling a __host__ function from a __host__ __device__ function is not allowed
```

It appears the issue is that on several lines in the file `KokkosSparse_sptrsv_solve_impl.hpp`, the code constructs a scalar with a brace initialisation.  If the ifpack2 code is instantiated for the scalar being the ensemble type, these end up being brace initialisations for ensembles.  The compilers seems to want to map these brace initialisations to the MP Vector initializer list constructor, which currently is not marked with a Kokkos macro.

**Comment by jhux2** [comment]
@trilinos/stokhos

**Comment by maartenarnst** [comment]
Closed in #11338.
